Build failure with GHC 9.2.1

```
src/SequenceFormats/Utils.hs:76:41: error:
    • Couldn't match type: forall x' x.
                           Pipes.Internal.Proxy x' x () PB.ByteString m0 ()
                     with: Pipes.Internal.Proxy
                             Pipes.Internal.X () () PB.ByteString m ()
      Expected: GHC.IO.Handle.Types.Handle
                -> Pipes.Internal.Proxy Pipes.Internal.X () () PB.ByteString m ()
        Actual: GHC.IO.Handle.Types.Handle
                -> Pipes.Core.Producer' PB.ByteString m0 ()
    • In the third argument of ‘PS.withFile’, namely ‘PB.fromHandle’
      In the expression: PS.withFile f ReadMode PB.fromHandle
      In an equation for ‘readFileProd’:
          readFileProd f = PS.withFile f ReadMode PB.fromHandle
    • Relevant bindings include
        readFileProd :: FilePath -> Producer PB.ByteString m ()
          (bound at src/SequenceFormats/Utils.hs:76:1)
   |
76 | readFileProd f = PS.withFile f ReadMode PB.fromHandle
   |                                         ^^^^^^^^^^^^^
cabal: Failed to build sequence-formats-1.6.1
```

--- a/src/SequenceFormats/FreqSum.hs
+++ b/src/SequenceFormats/FreqSum.hs
@@ -73,7 +73,7 @@ readFreqSumStdIn = readFreqSumProd PB.stdin
 
 -- |A function to read a freqsum file from a file. Returns a pair of a freqSum Header and a Producer over all lines.
 readFreqSumFile :: (MonadSafe m) => FilePath -> m (FreqSumHeader, Producer FreqSumEntry m ())
-readFreqSumFile file = readFreqSumProd $ withFile file ReadMode PB.fromHandle
+readFreqSumFile file = readFreqSumProd $ withFile file ReadMode (\h -> PB.fromHandle h)
 
 parseFreqSumHeader :: A.Parser FreqSumHeader
 parseFreqSumHeader = do

--- a/src/SequenceFormats/FreqSum.hs
+++ b/src/SequenceFormats/FreqSum.hs
@@ -7,7 +7,7 @@
 module SequenceFormats.FreqSum (readFreqSumStdIn, readFreqSumFile, FreqSumEntry(..),  
     FreqSumHeader(..), printFreqSumStdOut, printFreqSumFile, freqSumEntryToText) where
 
-import SequenceFormats.Utils (consumeProducer, Chrom(..))
+import SequenceFormats.Utils (consumeProducer, Chrom(..), readFileProd)
 
 import Control.Applicative ((<|>))
 import Control.Monad.Catch (MonadThrow, throwM)
@@ -73,7 +73,7 @@ readFreqSumStdIn = readFreqSumProd PB.stdin
 
 -- |A function to read a freqsum file from a file. Returns a pair of a freqSum Header and a Producer over all lines.
 readFreqSumFile :: (MonadSafe m) => FilePath -> m (FreqSumHeader, Producer FreqSumEntry m ())
-readFreqSumFile file = readFreqSumProd $ withFile file ReadMode (\h -> PB.fromHandle h)
+readFreqSumFile = readFreqSumProd . readFileProd
 
 parseFreqSumHeader :: A.Parser FreqSumHeader
 parseFreqSumHeader = do

--- a/src/SequenceFormats/Plink.hs
+++ b/src/SequenceFormats/Plink.hs
@@ -100,7 +100,7 @@ readPlinkBedProd nrInds prod = do
 
 -- |A function to read a bed file from a file. Returns a Producer over all lines.
 readPlinkBedFile :: (MonadSafe m) => FilePath -> Int -> m (Producer GenoLine m ())
-readPlinkBedFile file nrInds = readPlinkBedProd nrInds (PS.withFile file ReadMode PB.fromHandle)
+readPlinkBedFile file nrInds = readPlinkBedProd nrInds (PS.withFile file ReadMode (\h -> PB.fromHandle h))
 
 -- |Function to read a Bim File from StdIn. Returns a Pipes-Producer over the EigenstratSnpEntries.
 readBimStdIn :: (MonadThrow m, MonadIO m) => Producer EigenstratSnpEntry m ()

--- a/src/SequenceFormats/Plink.hs
+++ b/src/SequenceFormats/Plink.hs
@@ -100,7 +100,7 @@ readPlinkBedProd nrInds prod = do
 
 -- |A function to read a bed file from a file. Returns a Producer over all lines.
 readPlinkBedFile :: (MonadSafe m) => FilePath -> Int -> m (Producer GenoLine m ())
-readPlinkBedFile file nrInds = readPlinkBedProd nrInds (PS.withFile file ReadMode (\h -> PB.fromHandle h))
+readPlinkBedFile file nrInds = readPlinkBedProd nrInds (readFileProd file)
 
 -- |Function to read a Bim File from StdIn. Returns a Pipes-Producer over the EigenstratSnpEntries.
 readBimStdIn :: (MonadThrow m, MonadIO m) => Producer EigenstratSnpEntry m ()

--- a/src/SequenceFormats/RareAlleleHistogram.hs
+++ b/src/SequenceFormats/RareAlleleHistogram.hs
@@ -93,7 +93,7 @@ readHistogram path = liftIO $ withFile path ReadMode readHistogramFromHandle
 -- |Read a histogram from a File Handle.
 readHistogramFromHandle :: (MonadIO m) => Handle -> m RareAlleleHistogram
 readHistogramFromHandle handle = do
-    res <- evalStateT (parse parseHistogram) . (\h -> PB.fromHandle h) $ handle
+    res <- evalStateT (parse parseHistogram) (PB.fromHandle handle)
     case res of
         Nothing -> throw (SeqFormatException "histogram file exhausted too early")
         Just (Left err) -> throw (SeqFormatException ("Histogram parsing error: " <> show err))

--- a/src/SequenceFormats/RareAlleleHistogram.hs
+++ b/src/SequenceFormats/RareAlleleHistogram.hs
@@ -93,7 +93,7 @@ readHistogram path = liftIO $ withFile path ReadMode readHistogramFromHandle
 -- |Read a histogram from a File Handle.
 readHistogramFromHandle :: (MonadIO m) => Handle -> m RareAlleleHistogram
 readHistogramFromHandle handle = do
-    res <- evalStateT (parse parseHistogram) . PB.fromHandle $ handle
+    res <- evalStateT (parse parseHistogram) . (\h -> PB.fromHandle h) $ handle
     case res of
         Nothing -> throw (SeqFormatException "histogram file exhausted too early")
         Just (Left err) -> throw (SeqFormatException ("Histogram parsing error: " <> show err))

--- a/src/SequenceFormats/Utils.hs
+++ b/src/SequenceFormats/Utils.hs
@@ -73,7 +73,7 @@ consumeProducer :: (MonadThrow m) => A.Parser a -> Producer B.ByteString m () ->
 consumeProducer parser prod = parsed parser prod >>= liftParsingErrors
 
 readFileProd :: (PS.MonadSafe m) => FilePath -> Producer B.ByteString m ()
-readFileProd f = PS.withFile f ReadMode PB.fromHandle
+readFileProd f = PS.withFile f ReadMode (\h -> PB.fromHandle h)
 
 word :: A.Parser B.ByteString
 word = A.takeTill isSpace

